aptos-move/move-examples/fungible_asset/moon_coin/sources/MoonCoin.move

```diff
@@ -80,20 +82,12 @@ module example_addr::simple_managed_coin {
         fungible_asset::burn_from(burn_ref, from_wallet, amount);
     }
 
-    /// Freeze an account so it cannot transfer or receive fungible assets.
-    public entry fun freeze_account(admin: &signer, account: address) acquires ManagedFungibleAsset {
-        let asset = get_metadata();
-        let transfer_ref = &authorized_borrow_refs(admin, asset).transfer_ref;
-        let wallet = primary_fungible_store::ensure_primary_store_exists(account, asset);
-        fungible_asset::set_frozen_flag(transfer_ref, wallet, true);
-    }
-
-    /// Unfreeze an account so it can transfer or receive fungible assets.
-    public entry fun unfreeze_account(admin: &signer, account: address) acquires ManagedFungibleAsset {
+    /// Freeze/Unfreeze an account so it cannot transfer or receive fungible assets.
+    public entry fun set_frozen_flag(admin: &signer, account: address, frozen: bool) acquires ManagedFungibleAsset {
```

Comment:
I actually like having separate freeze and unfreeze functions for more expressiveness

Reply:
that's small nits. I could fix it